cocos/asset/assets/buffer-asset.ts

```diff
@@ -22,49 +22,51 @@
  THE SOFTWARE.
 */
 
-import { ccclass, override } from 'cc.decorator';
-import { assertIsNonNullable, cclegacy } from '../../core';
+import { ccclass, override, serializable } from 'cc.decorator';
+import { cclegacy } from '../../core';
 import { Asset } from './asset';
 
 /**
  * @en
  * `BufferAsset` is a kind of assets whose internal data is a section of memory buffer
- * that you can access through the [[BufferAsset.buffer]] function.
+ * that you can access through `BufferAsset.view`.
  * @zh
- * `BufferAsset` 是一类资产，其内部数据是一段内存缓冲，你可以通过 [[BufferAsset.buffer]] 函数获取其内部数据。
+ * `BufferAsset` 是一类资产，其内部数据是一段内存缓冲，你可以通过 `BufferAsset.view` 获取其内部数据。
  */
 @ccclass('cc.BufferAsset')
 export class BufferAsset extends Asset {
-    private _buffer: ArrayBuffer | null = null;
+    /**
+     * @zh 缓冲数据的字节视图。
+     * @en Byte view of the buffered data.
+     */
+    @serializable
+    public view = new Uint8Array();
 
     /**
-     * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
+     * @zh 首次调用将 **复制** 此时的`this.view`，并返回副本的 `ArrayBuffer`；该副本会在后续调用中直接返回。
+     * @en The first invocation on this method will **clone** `this.view` and returns `ArrayBuffer` of the copy.
+     * The copy will be directly returned in following invocations.
+     *
+     * @returns @en The `ArrayBuffer`. @zh `ArrayBuffer`。
+     *
+     * @deprecated @zh 自 3.9.0，此方法废弃，调用此方法将带来可观的性能开销；请转而使用 `this.view`。
+     * @en Since 3.9.0, this method is deprecated.
+     * Invocation on this method leads to significate cost. Use `this.view` instead.
      */
-    @override
-    get _nativeAsset (): ArrayBuffer | ArrayBufferView {
-        return this._buffer as ArrayBuffer;
```

Comment:
The overridden `get/set _nativeAsset` is removed. Will it break the compatibility?
The old `get _nativeAsset` return `this._buffer`, but now it returns `this._file` from its parent class `Asset`.

Reply:
As documented, this is an engine private interface and has been deprecated since v3.5.0 and is explicitly starts with an underscore. We can directly removed it in v3.9.0.